localization_manager/src/LocalizationTransitionTable.cpp

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <ros/console.h>
+#include "localization_manager/LocalizationTransitionTable.h"
+
+namespace localizer
+{
+LocalizationTransitionTable::LocalizationTransitionTable(LocalizerMode mode) : mode_(mode)
+{
+}
+LocalizationState LocalizationTransitionTable::getState() const
+{
+  return state_;
+}
+
+void LocalizationTransitionTable::logDebugSignal(LocalizationSignal signal) const
+{
+  ROS_DEBUG_STREAM("LocalizationTransitionTable received unsupported signal of " << signal << " while in state "
+                                                                                 << state_);
+}
+
+void LocalizationTransitionTable::setAndLogState(LocalizationState new_state, LocalizationSignal source_signal)
+{
+  if (new_state == state_)
+  {
+    return;  // State was unchanged no need to log or trigger callbacks
+  }
+  ROS_INFO_STREAM("LocalizationTransitionTable changed localization state from "
```

Comment:
Should we put up WARN messages for notable transitions like vehicle going into degraded, or unusable status?

Reply:
Since this node publishes the localization status anyway I think a single info log for all transitions is sufficient. No need to add extra if statements just to switch from INFO log to WARN log I think.